### PR TITLE
Add treatment of session authentication to `MaybeAuth`

### PR DESCRIPTION
**Pull request by ClaudivanFilho**
#### What is the purpose of this pull request?
- Add treatment of session authentication to `MaybeAuth`.

#### How should this be manually tested?
https://claudivan--storecomponents.myvtex.com/account/orders

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/MaybeAuth.tsx [hunk 1]
```diff
@@ -57,25 +58,32 @@ export default class MaybeAuth extends PureComponent<Props, State> {
...
+        .then(response => response.json())
+        .catch(() => null)
+      const promiseAuth = fetch(AUTH_STORE_URL, { credentials: 'same-origin' })
+        .then(response => response.json())
+        .catch(() => null)
+      const [sessionRes, authRes] = await Promise.all([
+        promiseSession, promiseAuth
+      ])
+      if (
+        (sessionRes && sessionRes.namespaces.profile.isAuthenticated.value !== 'False') ||
```
Comment (brunojdo) [inline react/components/MaybeAuth.tsx]: Create a func `convertToBool`, and return `toLower(sessionRes.namespaces.profile.isAuthenticated.value) === 'true' `

File: react/components/MaybeAuth.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, {PureComponent} from 'react'
 import Loading from './Loading'
 
 const LOGIN_PATH = '/login'
-const AUTH_STORE_URL = '/_v/private/authenticated/store'
+const API_SESSION_URL = '/api/sessions?items=*'
```
Comment (brunojdo) [inline react/components/MaybeAuth.tsx]: You need to make a `POST /api/sessions` with a header `Content-Type: application/json` to initialize session . (retrieve `vtex_session` and `vtex_segment` cookies)

Comment (brunojdo) [reply react/components/MaybeAuth.tsx]: PS: Check if you already have these cookies, if not make the request.

Comment (brunoabreu) [reply react/components/MaybeAuth.tsx]: Actually `vtex_session` is (or should be) an HTTP only cookie, so there is no way to know if there cookie is there using javascript.

**Pull request merged**